cloudimg/aws.py

```diff
@@ -486,10 +486,15 @@ def publish(self, metadata):
 
             if not snapshot:
                 log.info('Snapshot does not exist: %s', metadata.snapshot_name)
-                log.info('Searching for object: %s/%s',
-                         metadata.container, metadata.object_name)
-                obj = self.get_object_by_name(metadata.container,
-                                              metadata.object_name)
+                log.info(
+                    'Searching for object: %s/%s',
+                    metadata.container,
+                    metadata.object_name.rstrip(".xz"),
```

Comment:
So the object_name isn't actually the object name then?  Is it possible to fix that instead?

The way it's done now seems confusing, for example:

- the log message a few lines later is potentially false.  It's going to say "Object does not exist: foo.xz", while the code can't possibly know that, since it actually checked for "foo".
- there is a method called `get_object_by_name`, and a property called `object_name`, yet somehow the latter isn't the right value to pass into the former and instead needs some extra processing?  That seems like a trap for developers.

Reply:
AFAIK whenever the object name ends with `.xz` it will be decompressed as you [can see here](https://github.com/release-engineering/cloudimg/blob/master/cloudimg/aws.py#L423-L430). However, we need to call the `upload_to_container` method with the full name (with `xz`) in order to make the decompression.

With this I understand the `object_name` has to be the original name for `upload_to_container` but we need somehow to translate the decompression to the `get_object_by_name`...

I can also patch the later message which says "Object does not exist" but I agree it starts to behave like a trap for developers..

Another idea is to refactor the `upload_to_container` passing some arg like `xz_decompress = True` so it knows that the `object_name` is a xz compressed file and then we strip the extension just one time and pass the trimmed name everywhere... WDYT?

Reply:
Right, so I think the problem I'm talking about was already introduced by that earlier change to decompress .xz, because that's the change which introduced the concept of PublishingMetadata.object_name not matching the actual object name used in S3.

That being said, there is no documentation for PublishingMetadata.object_name, so I guess it can mean whatever one wants it to mean.  But I think the most reasonable expectation is for it to exactly match the S3 or Azure blob/object name.

I would say the right fix is that the earlier code added on upload:

```python
elif object_name.endswith(".xz"):
```

...should not be looking at object_name but instead at image_path.  Then object_name should be fixed so that it always matches the S3/Azure object name as was likely the original intent, e.g. the stripping of .xz happens inside of `def object_name`.

image_path is the input, object_name is the output.  If we are trying to make decisions about how to process the input (do we need to decompress it or not?), it makes more sense to look at the input filename rather than the output filename.  If anything, having an object_name ending in ".xz" would be more intuitively interpreted as "this is a request to *compress* my *output*" rather than "this is a request to *decompress* my *input*".

I'm not going to insist on a potentially risky refactoring though.  My minimum ask here would be just to remove any possibility that the following log message might be false, because lying log messages can cause people to waste a lot of time when investigating issues.

Reply:
I've refactored it with your suggestion, I believe it makes sense.

PTAL and let me know if it's ok for you.

Reply:
I think `def upload_to_container` is perfect now.  The other part of the problem -`PublishingMetadata.object_name` is not the real object name - is still there, so the refactoring didn't go as far as it could, it didn't implement this suggestion:

> Then object_name should be fixed so that it always matches the S3/Azure object name as was likely the original intent, e.g. the stripping of .xz happens inside of def object_name.

It's certainly an improvement though.

Reply:
Oh, you're right, now I got it!

I've adjusted it, PTAL again